usaspending_api/etl/management/commands/combine_universal_transaction_matview_chunks.py

```diff
@@ -0,0 +1,200 @@
+ALTER INDEX IF EXISTS idx_ut_simple_pop_geolocation RENAME TO idx_ut_simple_pop_geolocation_old;
+ALTER INDEX IF EXISTS idx_ut_recipient_hash RENAME TO idx_ut_recipient_hash_old;
+ALTER INDEX IF EXISTS idx_ut_action_date_pre2008 RENAME TO idx_ut_action_date_pre2008_old;
+
+ALTER TABLE universal_transaction_matview_temp RENAME TO universal_transaction_matview;
+
+ALTER INDEX idx_ut_transaction_id_temp RENAME TO idx_ut_transaction_id;
+ALTER INDEX idx_ut_action_date_temp RENAME TO idx_ut_action_date;
+ALTER INDEX idx_ut_last_modified_date_temp RENAME TO idx_ut_last_modified_date;
+ALTER INDEX idx_ut_fiscal_year_temp RENAME TO idx_ut_fiscal_year;
+ALTER INDEX idx_ut_type_temp RENAME TO idx_ut_type;
+ALTER INDEX idx_ut_award_id_temp RENAME TO idx_ut_award_id;
+ALTER INDEX idx_ut_pop_zip5_temp RENAME TO idx_ut_pop_zip5;
+ALTER INDEX idx_ut_recipient_unique_id_temp RENAME TO idx_ut_recipient_unique_id;
+ALTER INDEX idx_ut_parent_recipient_unique_id_temp RENAME TO idx_ut_parent_recipient_unique_id;
+ALTER INDEX idx_ut_simple_pop_geolocation_temp RENAME TO idx_ut_simple_pop_geolocation;
+ALTER INDEX idx_ut_recipient_hash_temp RENAME TO idx_ut_recipient_hash;
+ALTER INDEX idx_ut_action_date_pre2008_temp RENAME TO idx_ut_action_date_pre2008;
+"""
+
+REMOVE_OLD_TABLE_SQL = """
+DROP TABLE IF EXISTS universal_transaction_matview_old;
+"""
+
+REMOVE_MATVIEW_DATA_SQL = """
+REFRESH MATERIALIZED VIEW universal_transaction_matview_{current_chunk} WITH NO DATA;
+"""
+
+ANALYZE_TABLE_SQL = """
+ANALYZE VERBOSE universal_transaction_matview;
+"""
+
+TABLE_PERMISSION_SQL = """
+GRANT SELECT ON universal_transaction_matview TO readonly;
+"""
+
+
+class Command(BaseCommand):
+
+    help = """
+    This script combines the chunked Universal Transaction Matviews and
+    combines them into a single table.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument("--chunk-count", default=10, help="Number of chunked matviews to read from", type=int)
+        parser.add_argument("--analyze", action="store_true", help="Indicates whether table should be analyzed")
+        parser.add_argument(
+            "--keep-old-data",
+            action="store_true",
+            default=False,
+            help="Indicates whether or not to drop old table and matviews at end of command",
+        )
+
+    def handle(self, *args, **options):
+        chunk_count = options["chunk_count"]
+
+        logger.info(f"Chunk Count: {chunk_count}")
+
+        with Timer("Recreating table"):
+            self.recreate_matview()
+
+        with Timer("Inserting data into table"):
+            self.insert_matview_data(chunk_count)
+
+        with Timer("Creating table indexes"):
+            self.create_indexes()
+
+        with Timer("Swapping Tables/Indexes"):
+            self.swap_matviews()
+
+        if not options["keep_old_data"]:
+            self.remove_old_data(chunk_count)
+
+        if options["analyze"]:
```

Comment:
I can't think of a good reason to not analyze the new table after it is created and indexed. Perhaps it can always occur and the cli argument can be removed